src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -421,7 +442,7 @@ internal string GetLanguageForUri(Uri uri)
             languageId = trackedDocument.LanguageId;
         }
 
-        var documentFilePath = ProtocolConversions.GetDocumentFilePathFromUri(uri);
+        var documentFilePath = uri.Scheme == SourceGeneratedDocumentUris.Scheme ? uri.LocalPath : ProtocolConversions.GetDocumentFilePathFromUri(uri);
```

Comment:
I decided to put this check in GetDocumentFilePathFromUri, since that's where we have the special rule about how file:// URIs work versus other things in documents, and I absolutely would expect that a source generated doc could still land in miscellaneous files. Since this is specifically looking for a file path just to determine language name, this seemed fine.

Reply:
> I decided to put this check in GetDocumentFilePathFromUri

Decided to, or not to?  I think I would prefer if this check was handled in `GetDocumentFilePathFromUri`

Reply:
@dibarbet Re-reading my comment it made zero sense so I just rewrote it entirely. Let me know if that helps.

Reply:
@jasonmalinowski  Actually just noticed - do we even need to do this check?  I see this implementation for `GetDocumentFilePathFromUri`:
```
public static string GetDocumentFilePathFromUri(Uri uri)
    => uri.IsFile ? uri.LocalPath : uri.AbsoluteUri;
```

Which looks like it should handle the non-file case already

Reply:
So the non-file case means we use AbsoluteUri which includes the query string. Consider the case of two git files, say git:Foo.cs?version=1234 and git:Foo.cs?version=2345. We need to count those as separate documents in miscellaneous files, so it's correct that GetDocumentFilePathFromUri is using AbsolouteUri in this case.

But you are pointing to a deeper problem I think: GetLanguageForUri should never be calling AbsoluteUri anywhere (even indirectly) since that means if we take that path we're passing in the "?version=1234" into https://github.com/jasonmalinowski/roslyn/blob/f3da440d747e7ec80e97ed8f748dfda529bca23a/src/Features/LanguageServer/Protocol/LanguageInfoProvider.cs#L38 and that doesn't end well. So my change here is wrong, and the existing code is also wrong. GetLanguageForUri should always be using local path or something that drops the query string under the assumption the main part of the file has a useful extension. (and if that assumption isn't right then we're in trouble since at that point the URI itself is untrustworthy.)

Reply:
I also should take a second look at this -- I fixed this up as a part of some testing and I'm now seeing I probably shouldn't have even gotten there in the first place -- the initial check for URI + language name probably should have worked, I'd imagine? So maybe I have another bug somewhere.